napari/layers/labels/_labels_mouse_bindings.py

```diff
@@ -0,0 +1,61 @@
+from copy import copy
+
+from ._labels_utils import interpolate_coordinates
+
+
+def pick(layer, event):
+    """Change the selected label to the same as the region clicked."""
+    # on press
+    layer.selected_label = layer._value or 0
+    yield
+
+    # on move
+    while event.type == 'mouse_move':
+        yield
+
+    # on release
+    layer._last_cursor_coord = None
```

Comment:
this is something that I wanted to ask you on the original branch. I converted it to this because I wanted to avoid running into bugs from trying to clean the code up so I created the exact equivalent of what it was before

from my guess, the code in the on release section is specific to painting and can be removed from here (along with the on move section)

Reply:
great catch, yes this can be dropped